core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java

```diff
@@ -108,9 +112,40 @@ public List<CloudObjectLocation> getObjects()
   @Nullable
   @JsonProperty
   @JsonInclude(JsonInclude.Include.NON_NULL)
-  public String getFilter()
+  public String getObjectGlob()
   {
-    return filter;
+    return objectGlob;
+  }
+
+  /**
+   * Strips out blob store's protocol and bucket from {@link #getObjectGlob}.
+   * This is to reduce user errors when crafting a objectGlob.
```

Comment:
I think we shouldn't remove the bucket without verifying that it actually matches. This leads to surprising outcomes like the glob `s3://foo/*.txt` matching the file `s3://bar/x.txt`. I think we have a couple of options:

- Don't remove the protocol and bucket: if the user provides the glob `s3://foo/*.txt` then it matches nothing.
- Be "smart": apply the glob to the entire URI if it starts with a protocol; else apply it only to the object.

IMO, the first option is best. My experience with "smart" features is they end up being more confusing than they're worth.

Reply:
I agree with you, let's not be too clever.